pkgs/applications/editors/tiled/default.nix

```diff
@@ -1,19 +1,61 @@
-{ lib, mkDerivation, fetchFromGitHub, pkg-config, qmake
-, python3, qtbase, qttools }:
+{ lib
+, stdenv
+, env
+, fetchFromGitHub
+, pkg-config
+, qbs
+, wrapQtAppsHook
+, qtbase
+, qtdeclarative
+, qtsvg
+, qttools
+}:
 
-mkDerivation rec {
+let
+  qtEnv = env "tiled-qt-env" [ qtbase qtsvg qttools qtdeclarative ];
+in
+
+stdenv.mkDerivation rec {
   pname = "tiled";
-  version = "1.8.4";
+  version = "1.9.1";
 
   src = fetchFromGitHub {
     owner = "bjorn";
     repo = pname;
     rev = "v${version}";
-    sha256 = "sha256-QYA2krbwH807BkzVST+/+sjSR6So/aGY/YenEjYxE48=";
+    sha256 = "sha256-6vi9uuTsAoCdm5D14mqFYyTDDmCkQEA5mi1xRe0qy0w=";
   };
 
-  nativeBuildInputs = [ pkg-config qmake ];
-  buildInputs = [ python3 qtbase qttools ];
+  nativeBuildInputs = [ pkg-config qbs wrapQtAppsHook ];
+  buildInputs = [ qtEnv ];
+
+  outputs = [ "out" "dev" ];
+
+  configurePhase = ''
+    runHook preConfigure
+
+    qbs setup-qt --settings-dir . ${qtEnv}/bin/qmake qtenv
```

Comment:
I am not sure this will work correct with buildInputs set like this. Can you try building it with strictDeps set to true? If that fails cross compilation wouldn't work.

Reply:
It fails to build with strictDeps. qbs doesn't support multi-prefix qt setups so this workaround is needed.

Reply:
> It fails to build with strictDeps. 

How? This usually means cross compilation is broken.

Reply:
Fixed the build. Sorry i didn't read the whole log. It was missing a few dependencies with strictDeps.

Reply:
I don't think this would work when cross compiling. @jtojnar can you maybe help out?

Reply:
`qbs` uses `qmake` just for finding Qt installation prefix and does not respect `QMAKEPATH`. I setup `qbs` to use `qmake from `PATH` first but it fails with 

```
/build/source/dist/distribute.qbs:53:16 File '/nix/store/gg6b5gfmyhqk5l08bjswrhijzr1bfa1p-qtbase-5.15.5-dev/lib/libQt5Concurrent.so.5' does not exist.
/build/source/dist/distribute.qbs:53:16 File '/nix/store/gg6b5gfmyhqk5l08bjswrhijzr1bfa1p-qtbase-5.15.5-dev/lib/libQt5Concurrent.so.5.15.5' does not exist.
/build/source/dist/distribute.qbs:53:16 File '/nix/store/gg6b5gfmyhqk5l08bjswrhijzr1bfa1p-qtbase-5.15.5-dev/lib/libQt5Core.so.5' does not exist.
/build/source/dist/distribute.qbs:53:16 File '/nix/store/gg6b5gfmyhqk5l08bjswrhijzr1bfa1p-qtbase-5.15.5-dev/lib/libQt5Core.so.5.15.5' does not exist.
/build/source/dist/distribute.qbs:53:16 File '/nix/store/gg6b5gfmyhqk5l08bjswrhijzr1bfa1p-qtbase-5.15.5-dev/lib/libQt5Gui.so.5' does not exist.
/build/source/dist/distribute.qbs:53:16 File '/nix/store/gg6b5gfmyhqk5l08bjswrhijzr1bfa1p-qtbase-5.15.5-dev/lib/libQt5Gui.so.5.15.5' does not exist.
/build/source/dist/distribute.qbs:53:16 File '/nix/store/gg6b5gfmyhqk5l08bjswrhijzr1bfa1p-qtbase-5.15.5-dev/lib/libQt5Network.so.5' does not exist.
/build/source/dist/distribute.qbs:53:16 File '/nix/store/gg6b5gfmyhqk5l08bjswrhijzr1bfa1p-qtbase-5.15.5-dev/lib/libQt5Network.so.5.15.5' does not exist.
/build/source/dist/distribute.qbs:53:16 File '/nix/store/gg6b5gfmyhqk5l08bjswrhijzr1bfa1p-qtbase-5.15.5-dev/lib/libQt5Qml.so.5' does not exist.
/build/source/dist/distribute.qbs:53:16 File '/nix/store/gg6b5gfmyhqk5l08bjswrhijzr1bfa1p-qtbase-5.15.5-dev/lib/libQt5Qml.so.5.15.5' does not exist.
/build/source/dist/distribute.qbs:53:16 File '/nix/store/gg6b5gfmyhqk5l08bjswrhijzr1bfa1p-qtbase-5.15.5-dev/lib/libQt5Svg.so.5' does not exist.
/build/source/dist/distribute.qbs:53:16 File '/nix/store/gg6b5gfmyhqk5l08bjswrhijzr1bfa1p-qtbase-5.15.5-dev/lib/libQt5Svg.so.5.15.5' does not exist.
/build/source/dist/distribute.qbs:53:16 File '/nix/store/gg6b5gfmyhqk5l08bjswrhijzr1bfa1p-qtbase-5.15.5-dev/lib/libQt5Widgets.so.5' does not exist.
/build/source/dist/distribute.qbs:53:16 File '/nix/store/gg6b5gfmyhqk5l08bjswrhijzr1bfa1p-qtbase-5.15.5-dev/lib/libQt5Widgets.so.5.15.5' does not exist.
/build/source/dist/distribute.qbs:53:16 File '/nix/store/gg6b5gfmyhqk5l08bjswrhijzr1bfa1p-qtbase-5.15.5-dev/lib/libQt5DBus.so.5' does not exist.
/build/source/dist/distribute.qbs:53:16 File '/nix/store/gg6b5gfmyhqk5l08bjswrhijzr1bfa1p-qtbase-5.15.5-dev/lib/libQt5DBus.so.5.15.5' does not exist.
/build/source/dist/distribute.qbs:53:16 File '/nix/store/gg6b5gfmyhqk5l08bjswrhijzr1bfa1p-qtbase-5.15.5-dev/lib/libQt5XcbQpa.so.5' does not exist.
/build/source/dist/distribute.qbs:53:16 File '/nix/store/gg6b5gfmyhqk5l08bjswrhijzr1bfa1p-qtbase-5.15.5-dev/lib/libQt5XcbQpa.so.5.15.5' does not exist.
```

It thinks all Qt files are installed in `dev` prefix and tries to find them. `qbs` doesn't support multi-prefix Qt installations like we have.

Reply:
> It is not clear to me where does `env` come from


It's defined [here](https://github.com/NixOS/nixpkgs/blob/nixos-unstable/pkgs/development/libraries/qt-5/qt-env.nix).

Reply:
@muscaln Note that these errors are from `distribute.qbs`, which is a file maintained as part of Tiled and may not indicate an issue with Qbs. The file is also not built by default and not actually relevant when compiling or installing Tiled, so I'm not sure why those warnings show up (do they cause the build to fail?). `distribute.qbs` is only relevant when building a binary .zip release of Tiled.

Thanks a lot for updating the Tiled package, btw, especially dealing with the qmake to Qbs transition!